vinyldns/client.go

```diff
@@ -22,15 +23,22 @@ type ClientConfiguration struct {
 	AccessKey string
 	SecretKey string
 	Host      string
+	UserAgent string
 }
 
 // NewConfigFromEnv creates a new ClientConfiguration
 // using environment variables.
 func NewConfigFromEnv() ClientConfiguration {
+	ua := defaultUA()
+
+	if os.Getenv("VINYLDNS_USER_AGENT") != "" {
```

Comment:
Stylistic suggestions:
1) 
```
if vua := os.Getenv("VINYLDNS_USER_AGENT"); vua != "" {
    ua = vua
}
```
2) Since the string "VINYLDNS_USER_AGENT" is used in various parts of our code, it looks like a good candidate for a constant.